### ISSUE TITLE
[MD051/link-fragments] Feature Request: Add support for checking fragments in other files

### ISSUE THREAD

**Issue by Cameronsplaze** [DavidAnson/markdownlint#1533]
Checking link fragments is currently working great if you're linking to another place in the same file, but fails to warn if the target is in another file.

Similar to https://github.com/DavidAnson/markdownlint/issues/513#issuecomment-1085283870:

- **article.md**

  ```md
  You can see authors in [authors](readme.md#authors-this-is-a-typo) section in readme
  ```

- **readme.md**

  ```md
  # Authors
  
  - John Doe
  - Bob Foo
  ```

This will fail to warn that the readme `#authors-this-is-a-typo` header is invalid. Is there a way around this I'm missing maybe?

Thank you!

**Comment by DavidAnson** [comment]
This is by design and implied by the rule documentation that refers to links within a file. Scanning referenced files as you suggest would be possible, but is out of scope for this rule because it can't be done synchronously (or at all in some scenarios).

<https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md>

**Comment by Cameronsplaze** [comment]
Ah that makes sense, thanks!

Would you consider a feature request for adding this as a new rule then? Or is synchronously a blocker for rules?

**Comment by DavidAnson** [comment]
It's a blocker for core rules. Too much unpredictability for scenarios like VS Code in the web, etc.. But it's something someone could implement in a custom rule. Scanning quickly, I think one or two here may do what you want already: https://www.npmjs.com/search?q=keywords:markdownlint-rule
